modules/javascript/src/main/webapp/javascript/atmosphere.js

```diff
@@ -196,6 +203,16 @@
                 handleOnlineOffline: true,
                 maxWebsocketErrorRetries: 1,
                 curWebsocketErrorRetries: 0,
+                id: undefined,
+                openId: undefined,
+                reconnectId: undefined,
+                firstMessage: undefined,
+                isOpen: undefined,
+                isReopen: undefined,
+                closed: undefined,
+                ctime: undefined,
+                heartbeatTimer: undefined,
+                force: undefined,
```

Comment:
All these, plus `mrequest` and `webSocketUrl` above, were accessed as properties of the `_request` object (or copies thereof) but never declared. It is not strictly necessary, but helps to have an overview of all the properties that will be used.